src/Stratis.Bitcoin.Tests/Utilities/AsyncConsumerQueueTest.cs

```diff
@@ -0,0 +1,140 @@
+
+namespace Stratis.Bitcoin.Tests.Utilities
+{
+    public class AsyncConsumerQueueTest
+    {
+        [Fact]
+        public async void QueueReactsToTriggerAsync()
+        {
+            int summ = 0;
+
+            AsyncConsumerQueue<int> processedQueue = new AsyncConsumerQueue<int>(
+                innerQueue =>
+                {
+                    while (innerQueue.TryDequeue(out int item))
+                        summ += item;
+                }, item =>
+                {
+                    // Triger if '5' gets added.
+                    return item == 5;
+                } );
+
+            for (int i = 0; i < 20; i++)
+            {
+                processedQueue.Enqueue(i);
+                await Task.Delay(50);
+            }
+
+            processedQueue.Dispose();
+
+            // 0+1+2+3+4+5 == 15
+            Assert.Equal(15, summ);
+        }
+
+        [Fact]
+        public async void QueueReactsToTimerAsync()
+        {
+            int summ = 0;
+
+            AsyncConsumerQueue<int> processedQueue = new AsyncConsumerQueue<int>(
+                innerQueue =>
+                {
+                    while (innerQueue.TryDequeue(out int item))
+                        summ += item;
+                }, null,
+                TimeSpan.FromMilliseconds(500));
+
+            for (int i = 0; i < 5; i++)
+            {
+                processedQueue.Enqueue(i);
+                await Task.Delay(50);
+            }
+
+
+            // 250 ms after the start no items were processed.
+            Assert.Equal(5, processedQueue.Count);
+
+            await Task.Delay(300);
+
+            Assert.Equal(0, processedQueue.Count);
+            Assert.Equal(10, summ);
+            processedQueue.Dispose();
+        }
+
+        [Fact]
+        public async void QueueReactsToTriggerAndTimerAsync()
+        {
+            int summ = 0;
+
+            AsyncConsumerQueue<int> processedQueue = new AsyncConsumerQueue<int>(
+                innerQueue =>
+                {
+                    while (innerQueue.TryDequeue(out int item))
+                        summ += item;
+                }, item =>
+                {
+                    // Triger if '5' gets added.
+                    return item == 5;
+                }, TimeSpan.FromMilliseconds(500));
+
+            for (int i = 0; i < 6; i++)
+            {
+                processedQueue.Enqueue(i);
+                await Task.Delay(30);
+            }
+
+            // 15 after being triggered.
+            Assert.Equal(15, summ);
+
+
+            for (int i = 6; i < 10; i++)
+            {
+                processedQueue.Enqueue(i);
+                await Task.Delay(30);
+            }
+
+            await Task.Delay(500);
+
+            // 45 after a timer.
+            Assert.Equal(45, summ);
+            processedQueue.Dispose();
+        }
+
+        [Fact]
+        public async void TimerRestartsIfNotAllItemsWereConsumedAsync()
+        {
+            int summ = 0;
```

Comment:
i belive standard abbreviation for summation is just sum with one m

Reply:
Fixed.